src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleThreadContext.java

```diff
@@ -269,12 +269,16 @@ public InterimModule buildModule(@Nullable Registry registry) throws EvalExcepti
   }
 
   public ImmutableMap<String, ModuleOverride> buildOverrides() {
+    LinkedHashMap<String, ModuleOverride> effectiveOverrides = new LinkedHashMap<>();
+    if (!shouldIgnoreDevDeps()) {
```

Comment:
can we do this in `addOverride` instead? seems a bit more natural (and symmetric with dev deps handling)